engine/src/main/java/org/terasology/network/ServerPingSystem.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.network;
+
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.network.events.DeactivatePingServerEvent;
+import org.terasology.network.events.DisconnectedEvent;
+import org.terasology.network.events.PingFromClientEvent;
+import org.terasology.network.events.PingFromServerEvent;
+import org.terasology.registry.In;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+
+/**
+ * This system implement the server ping to clients on need base.
+ * It runs on the server, pings to all clients who subscribe this function.
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ServerPingSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+
+    private static final long pingPeriod = 10000;
```

Comment:
Just noticed this: Wouldn't it be cooler if it would be a lower value like 200?

To make it real time you could change the initialise method of OnlinePlayersOverlay to:
```
        text.bindText(new ReadOnlyBinding<String>() {
            @Override
            public String get() {
                PingStockComponent pingStockComp = localPlayer.getClientEntity().getComponent(PingStockComponent.class);
                if (pingStockComp == null) {
                    String playerListText = determinePlayerListText();
                    return playerListText;
                } else {
                    String playerAndPing = determinePlayerAndPing(pingStockComp);
                    return playerAndPing;
                }
            }
        });
```
instead of setting the text during the opening. During the opening it could subscribe.to pings and during closing it could unsubscribe.